src/sentry/static/sentry/app/components/projects/missingProjectMembership.tsx

```diff
@@ -47,101 +52,170 @@ class MissingProjectMembership extends React.Component<Props, State> {
       this.props.api,
       {
         orgId: this.props.organization.slug,
-        teamId: team.slug,
+        teamId: team.value,
       },
       {
         success: () => {
           this.setState({
             loading: false,
             error: false,
           });
+          addSuccessMessage(t('Request to join team sent.'));
         },
         error: () => {
           this.setState({
             loading: false,
             error: true,
           });
-          addErrorMessage(t('There was an error while trying to leave the team.'));
+          addErrorMessage(t('There was an error while trying to request access.'));
         },
       }
     );
   }
 
-  renderJoinTeam(team: Team, features: Set<string>) {
+  renderJoinTeam(team: string, features: Set<string>) {
+    const teamStatus = TeamStore.getBySlug(team);
+
     if (!team) {
       return null;
     }
     if (this.state.loading) {
-      return <a className="btn btn-default btn-loading btn-disabled">...</a>;
-    } else if (team.isPending) {
-      return <a className="btn btn-default btn-disabled">{t('Request Pending')}</a>;
+      if (features.has('open-membership')) {
+        return <StyledButton busy>Join Team</StyledButton>;
+      }
+      return <StyledButton busy>Request Access</StyledButton>;
+    } else if (teamStatus?.isPending) {
+      return <StyledButton disabled>Request Pending</StyledButton>;
     } else if (features.has('open-membership')) {
       return (
-        <a className="btn btn-default" onClick={this.joinTeam.bind(this, team)}>
+        <StyledButton
+          priority="primary"
+          type="button"
+          onClick={this.joinTeam.bind(this, team)}
```

Comment:
`team` here is the team slug, I think renaming it should help. I wonder why typescript did not catch this.